src/EventSourcingTests/Projections/Flattened/flat_table_projection_with_event_member_identifier_end_to_end.cs

```diff
@@ -0,0 +1,210 @@
+using System;
+using System.Data.Common;
+using System.Linq;
+using System.Threading.Tasks;
+using Baseline;
+using Marten.Events;
+using Marten.Events.Projections;
+using Marten.Testing.Harness;
+using Npgsql;
+using Shouldly;
+using Weasel.Core;
+using Weasel.Postgresql.Tables;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace EventSourcingTests.Projections.Flattened;
+
+public class flat_table_projection_with_event_member_identifier_end_to_end : OneOffConfigurationsContext
+{
+    private readonly ITestOutputHelper _output;
+
+    public flat_table_projection_with_event_member_identifier_end_to_end(ITestOutputHelper output)
+    {
+        _output = output;
+        StoreOptions(opts =>
+        {
+            opts.Projections.Add<WriteTableWithEventMemberIdentityProjection>(ProjectionLifecycle.Inline);
+            opts.Events.StreamIdentity = StreamIdentity.AsString;
+        }, true);
+    }
+
+    [Fact]
+    public async Task table_should_be_built()
+    {
+        await theStore.Storage.ApplyAllConfiguredChangesToDatabaseAsync();
+
+        using var conn = theStore.Storage.Database.CreateConnection();
+        await conn.OpenAsync();
+
+        var table = await new Table(new DbObjectName(SchemaName, "member_values")).FetchExisting(conn);
+
+        table.PrimaryKeyColumns.Single().ShouldBe("name");
+        table.Columns.Select(x => x.Name).OrderBy(x => x)
+            .ShouldHaveTheSameElementsAs("a", "b", "c", "d", "name", "revision", "status");
+    }
+
+    public class Data
+    {
+        public int A { get; set; }
+        public int B { get; set; }
+        public int C { get; set; }
+        public int D { get; set; }
+        public string Status { get; set; }
+        public int Version { get; set; }
+    }
+
+    private async Task<Data> readData(DbDataReader reader)
+    {
+        return new Data
+        {
+            A = await reader.GetFieldValueAsync<int>(1),
+            B = await reader.GetFieldValueAsync<int>(2),
+            C = await reader.GetFieldValueAsync<int>(3),
+            D = await reader.GetFieldValueAsync<int>(4),
+            Status = await reader.GetFieldValueAsync<string>(5),
+            Version = await reader.GetFieldValueAsync<int>(6)
+        };
+    }
+
+    private async Task<Data> findData(string name)
+    {
+        using var conn = theStore.Storage.Database.CreateConnection();
+        await conn.OpenAsync();
+
+        var all = await conn.CreateCommand($"select * from {SchemaName}.member_values where name = :id")
+            .With("id", name)
+            .FetchList(readData);
```

Comment:
**SUGGESTION:** That's later on some potential for convention-based mapping with code generation.

Reply:
Something like what LLBLGen is today? Or do you mean something else? This was strictly for testing. Dapper would have helped, but I'm not a big Dapper guy.